### DISCUSSION TITLE
Hard Limit and Alarm

### DISCUSSION THREAD

**Discussion by Switch-Witch** [terjeio/ioSender#415] [Q&A]
Hi, sorry for being so annoying and asking questions day by day....
I want to use hard limits to keep the machine safe inside the travel area.
When activating hard limits everything works as expected - I travels and stops at the max travel distance.
So far so good...
But when activated the hard limits every 15 seconds without any action IOSender turns into an Alarm Status (Alarm1-error9).
After resetting and unlocking everything works fine again until the next 15 seconds without action...

I tried out different settings but without success.

What could be the reason and how to avoid the alarm?
I really would like to use the hard limits.

thx in advance

best 

Kai

**Comment by terjeio** [comment]
This is typically caused by EMI (electromagnetic interference).
Are the limit LEDs blinking randomly?
And is the alarm happening even when there is no motion and the spindle is not running?

**Comment by Switch-Witch** [comment]
No, the limit LED`s are not blinking randomly.
The alarm happens when there is no motion and the spindle is not running.

**Reply by phil-barrett** [reply]
Terje, do short limit input pulses get stretched? If not, then there probably would not be any blinking.

Switch-Witch, please describe your limit switches - switch type, wire type and anything else.  Also, you may have said in another post but could you refresh us on what board you are using.

**Reply by terjeio** [reply]
> Terje, do short limit input pulses get stretched?

Not for the "LED"s. However, there are latches for input signals that can be output by the `$LEV` command (Last EVents). The limit switch latch will be reset by homing the machine. If one or more of the axis letters shows up in the output from this command the corresponding limit inputs was triggered, e.g if X limit has been triggered since last homing:
```
$lev
[LASTEVENTS:,X,,,]
```

**Reply by Switch-Witch** [reply]
Hi Phil,
> Switch-Witch, please describe your limit switches - switch type, wire type and anything else. Also, you may have said in another post but could you refresh us on what board you are using.
I am using your board(s) (3x) 😄 for sure...
I use inductive switches wich are wired with their own cables to a multipin - from the multipin to the board I did not use any special kind of shielded cables as the cables from the switches are also not shielded.

**Reply by phil-barrett** [reply]
Oh yeah, now I remember... Sorry.  I had knee surgery on Tuesday and am a bit light-headed from the analgesics.

I'd like to look at the multipin to see if it passing the signals.  Maybe a photo?
